cw_lightning/lib/lightning_wallet.dart

```diff
@@ -0,0 +1,397 @@
+
+  LightningWalletBase({
+    required String mnemonic,
+    required String password,
+    required WalletInfo walletInfo,
+    required Box<UnspentCoinsInfo> unspentCoinsInfo,
+    required Uint8List seedBytes,
+    String? addressPageType,
+    List<BitcoinAddressRecord>? initialAddresses,
+    LightningBalance? initialBalance,
+    Map<String, int>? initialRegularAddressIndex,
+    Map<String, int>? initialChangeAddressIndex,
+  })  : _isTransactionUpdating = false,
+        syncStatus = NotConnectedSyncStatus(),
+        _balance = ObservableMap<CryptoCurrency, LightningBalance>(),
+        mnemonic = mnemonic,
+        super(
+          password: password,
+          walletInfo: walletInfo,
+          unspentCoinsInfo: unspentCoinsInfo,
+          networkType: bitcoin.bitcoin,
+          initialAddresses: initialAddresses,
+          initialBalance: initialBalance,
+          seedBytes: seedBytes,
+          currency: CryptoCurrency.btcln,
+        ) {
+    _balance[CryptoCurrency.btcln] =
+        initialBalance ?? LightningBalance(confirmed: 0, unconfirmed: 0, frozen: 0);
+    String derivationPath = walletInfo.derivationInfo!.derivationPath!;
+    String sideDerivationPath = derivationPath.substring(0, derivationPath.length - 1) + "1";
+    final hd = bitcoin.HDWallet.fromSeed(seedBytes, network: networkType);
+    walletAddresses = BitcoinWalletAddresses(
+      walletInfo,
+      initialAddresses: initialAddresses,
+      initialRegularAddressIndex: initialRegularAddressIndex,
+      initialChangeAddressIndex: initialChangeAddressIndex,
+      mainHd: hd.derivePath(derivationPath),
+      sideHd: hd.derivePath(sideDerivationPath),
+      network: network,
+    );
+
+    // initialize breez:
+    try {
+      setupBreez(seedBytes);
+    } catch (e) {
+      print("Error initializing Breez: $e");
+    }
+
+    autorun((_) {
+      this.walletAddresses.isEnabledAutoGenerateSubaddress = this.isEnabledAutoGenerateSubaddress;
+    });
+  }
+
+  late final ObservableMap<CryptoCurrency, LightningBalance> _balance;
+  StreamSubscription<List<Payment>>? _paymentsSub;
+  StreamSubscription<NodeState?>? _nodeStateSub;
+
+  @override
+  @computed
+  ObservableMap<CryptoCurrency, LightningBalance> get balance => _balance;
+
+  static Future<LightningWallet> create(
+      {required String mnemonic,
+      required String password,
+      required WalletInfo walletInfo,
+      required Box<UnspentCoinsInfo> unspentCoinsInfo,
+      String? addressPageType,
+      List<BitcoinAddressRecord>? initialAddresses,
+      LightningBalance? initialBalance,
+      Map<String, int>? initialRegularAddressIndex,
+      Map<String, int>? initialChangeAddressIndex}) async {
+    return LightningWallet(
+      mnemonic: mnemonic,
+      password: password,
+      walletInfo: walletInfo,
+      unspentCoinsInfo: unspentCoinsInfo,
+      initialAddresses: initialAddresses,
+      initialBalance: initialBalance,
+      seedBytes: await Mnemonic.toSeed(mnemonic),
```

Comment:
this will work for bip39 seeds only, but we allow it for both bip39 and electrum.
so you either have to do it like `bitcoin_wallet.dart` but I guess we don't have the `derivationType` for Lightning, so you might want to check for the validate functions and decide upon their result which function to use to get the seedbytes